src/components/header/renderer.tsx

```diff
@@ -242,6 +249,28 @@ class Header extends React.Component<Props, State> {
     dlButton.classList.remove('disabled');
   }
 
+  public downloadJSON(event) {
+    if (
+      event.target &&
+      (event.target.matches('input') ||
+        event.target.matches('label') ||
+        event.target.matches('div.type-input-container'))
+    ) {
+      return;
+    }
+    const content = this.state.downloadVegaJSON ? this.props.vegaSpec : this.props.vegaLiteSpec;
+    const filename = this.state.downloadVegaJSON ? 'visualization.vg.json' : 'visualization.vl.json';
```

Comment:
In a follow up, please use template strings.

Reply:
Do they serve any benefit here?

Reply:
Yes, more readable code.